components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -360,6 +368,18 @@ private Application createOAuthApplication(ApplicationRegistrationRequest regist
             throw DCRMUtils.generateClientException(DCRMConstants.ErrorMessages.CONFLICT_EXISTING_CLIENT_ID,
                     registrationRequest.getConsumerKey());
         }
+        //validate SSA
+        if (StringUtils.isNotEmpty(registrationRequest.getSoftwareStatement())) {
+            try {
+                if (validateSSASignature(registrationRequest.getSoftwareStatement())) {
+                    throw new DCRMClientException(DCRMConstants.ErrorCodes.INVALID_SOFTWARE_STATEMENT,
+                            DCRMConstants.ErrorMessages.SIGNATURE_VALIDATION_FAILED.getMessage());
+                }
+            } catch (IdentityOAuth2Exception e) {
+                throw new DCRMClientException(DCRMConstants.ErrorCodes.INVALID_SOFTWARE_STATEMENT,
```

Comment:
e is missing and is it ok to send a client exception here?

Reply:
fixed in 6ee33bbad29564b167d3a5cd953cb081c4755c0f